src/plot_api/to_image.js

```diff
@@ -0,0 +1,95 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+/*eslint dot-notation: [2, {"allowPattern": "^catch$"}]*/
+
+'use strict';
+
+var Plotly = require('../plotly');
+
+/**
+ * @param {object} gd figure Object
+ * @param {object} opts option object
+ * @param opts.format 'jpeg' | 'png' | 'webp' | 'svg'
+ */
+function toImage(gd, opts) {
+    var Snapshot = require('../snapshot');
```

Comment:
embarrassed to say, but when I do this `Snapshot` is empty, and I don't understand why.  What I understand about scoping says that this makes no sense.  Defining as before and where you suggest should make it available within, but for some reason it is an empty `{}` when I put it where it belongs.